### PR TITLE
Added custom metric for patient upload

### PR DESCRIPTION
**Pull request by johanna-skylight**
# FRONTEND PULL REQUEST

## Related Issue

We want to measure how long it takes for a user to complete an upload request with the new patient bulk upload feature

## Changes Proposed

Added logic to create a custom metric in azure. We use the native performance API to make the calculations of how long it takes from the user landing in the page to the point it gets a successful response from the server

## Testing

Test in any dev environment that the metric is successfully sent to azure 

## Screenshots / Demos
Screenshot 2023-01-23 at 3 59 50 PM
Screenshot 2023-01-23 at 3 50 51 PM

### CODE REVIEW

**Review by emmastephenson — APPROVED** [review]
Just to clarify though, this won't record the total time taken to upload patients, right? Since that piece is done in an asynchronous thread.

**Comment by emmastephenson** [comment]
Also - would it be possible to separately track pageviews to this page? Definitely don't have to include this here, just wondering how complex that would be.

**Review by BobanL — APPROVED** [review]

**Comment by johanna-skylight** [comment]
> Also - would it be possible to separately track pageviews to this page? Definitely don't have to include this here, just wondering how complex that would be.

This page is already being tracked by default but because the title of the page doesn't change for the different add patient options both get tracked under page add patient, you can differentiate by operation name which is the url. 

If you want I can change the title of the page to be "add patient | upload" so the metric looks cleaner

File: frontend/src/app/patients/UploadPatients.tsx [hunk 1]
```diff
@@ -56,6 +60,16 @@ const UploadPatients = () => {
     facilities.find((f) => f.id === activeFacilityId) ||
     facilities[0] || { id: "", name: "" };
 
+  useEffect(() => {
+    performance.mark("patientUpload_pageLanding");
+
+    return () => {
+      performance.clearMarks("patientUpload_pageLanding");
```
Comment (emyl3) [inline frontend/src/app/patients/UploadPatients.tsx:67]: TIL! https://developer.mozilla.org/en-US/docs/Web/API/PerformanceEntry

**Review by emyl3 — APPROVED** [review]
Very nice!! Thanks for adding this and helping me learn something new! 🎉

**Pull request merged**